cfme/tests/infrastructure/test_timelines.py

```diff
@@ -3,163 +3,122 @@
-    navigate_to(InfraProvider, 'All')
-    vm = VM.factory(vm_name, provider)
+def test_vm(request, a_provider):
+    vm_name = "test_tl_" + fauxfactory.gen_alphanumeric(length=4)
+    vm = VM.factory(vm_name, a_provider)
 
     request.addfinalizer(vm.delete_from_provider)
 
-    if not provider.mgmt.does_vm_exist(vm_name):
-        vm.create_on_provider(find_in_cfme=True, allow_skip="default")
-        vm.refresh_relationships()
+    if not a_provider.mgmt.does_vm_exist(vm.name):
+        logger.info("deploying %s on provider %s", vm.name, a_provider.key)
+        vm.create_on_provider(allow_skip="default", find_in_cfme=True)
     return vm
 
 
 @pytest.fixture(scope="module")
-def gen_events(delete_fx_provider_event, provider, test_vm):
+def gen_events(test_vm):
     logger.debug('Starting, stopping VM')
-    mgmt = provider.mgmt
+    mgmt = test_vm.provider.mgmt
     mgmt.stop_vm(test_vm.name)
     mgmt.start_vm(test_vm.name)
 
 
-def count_events(vm, nav_step):
-    try:
-        nav_step()
-    except ToolbarOptionGreyedOrUnavailable:
-        return 0
-    except NoSuchElementException:
-        vm.rediscover()
-        return 0
-
-    events = []
-    for event in jstimelines.events():
-        data = event.block_info()
-        if vm.name in data.values():
-            events.append(event)
-            if len(events) > 0:
-                return len(events)
-    return 0
-
-
-@pytest.mark.uncollectif(lambda: version.current_version() >= '5.7')
+def count_events(target, vm):
+    timelines_view = navigate_to(target, 'Timelines')
+    timelines_view.filter.time_position.select_by_visible_text('centered')
+    timelines_view.filter.apply.click()
+    found_events = []
+    for evt in timelines_view.chart.get_events():
+        if not hasattr(evt, 'source_vm'):
+            # BZ(1428797)
+            logger.warn("event {evt} doesn't have source_vm field. Probably issue".format(evt=evt))
+            continue
+        elif evt.source_vm == vm.name:
+            found_events.append(evt)
+
+    logger.info("found events: {evt}".format(evt="\n".join([repr(e) for e in found_events])))
+    return len(found_events)
+
+
+@pytest.mark.uncollectif(lambda: version.current_version() < '5.7')
 @pytest.mark.meta(blockers=[1264183, 1281746])
-def test_provider_event(provider, gen_events, test_vm):
+def test_provider_event(gen_events, test_vm):
     """Tests provider event on timelines
 
     Metadata:
         test_flag: timelines, provision
     """
-    def nav_step():
-        navigate_to(provider, 'Details')
 
-        toolbar.select('Monitoring', 'Timelines')
-    wait_for(count_events, [test_vm, nav_step], timeout='5m', fail_condition=0,
+    wait_for(count_events, [test_vm.provider, test_vm], timeout='5m', fail_condition=0,
              message="events to appear")
 
 
-@pytest.mark.uncollectif(lambda: version.current_version() >= '5.7')
+@pytest.mark.uncollectif(lambda: version.current_version() < '5.7')
 @pytest.mark.meta(blockers=[1281746])
```

Comment:
While you're here, this bug is closed, can probably remove. Why the switch from `>=` to `<`?